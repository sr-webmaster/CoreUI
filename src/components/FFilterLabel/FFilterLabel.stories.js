import { MAIN } from '../../../.storybook/categories'
import FFilterLabel from './FFilterLabel.vue'

export default {
  title: `${MAIN}|FFilterLabel`,
  id: 'FFilterLabel',
  decorators: [
    () => ({
      template: `
        <v-container class="grey"><story /></v-container>
      `
    })
  ]
}

/**
 * https://storybook.js.org/docs/guides/guide-vue/#step-4-write-your-stories
 */
export const defaultStory = () => ({

  components: {
    FFilterLabel
  },
  template: `
    <f-filter-label>Label</f-filter-label>
  `
})

export const alternativeColor = () => ({
  components: {
    FFilterLabel
  },
  template: `
    <f-filter-label color="black">Label</f-filter-label>
  `
})
