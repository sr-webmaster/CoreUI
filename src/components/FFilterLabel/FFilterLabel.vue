<template>
  <div :class="`${color}--text`">
    <span class="no-wrap font-weight-bold">
      <slot />
    </span>
  </div>
</template>

<script>
export default {
  props: {
    color: {
      type: String,
      default: 'white'
    }
  }
}
</script>

<style>
  .no-wrap{
    white-space: nowrap;
    text-overflow: ellipsis;
    overflow: hidden;
  }
</style>
