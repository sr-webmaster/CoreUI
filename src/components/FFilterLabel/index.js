import Component from './FFilterLabel'
export default Component
