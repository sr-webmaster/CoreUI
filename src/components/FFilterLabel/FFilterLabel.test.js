import { mount, shallowMount } from '@vue/test-utils'
import createLocalVue from 'vue-cli-plugin-freshinup-ui/utils/testing/createLocalVue'
import Component from '@freshinup/core-ui/src/components/FFilterLabel'
import { defaultStory, alternativeColor } from './FFilterLabel.stories'

describe('FFilterLabel', () => {
  describe('Computed', () => {
    // Using shallowMount test the computed properties
    const vm = shallowMount(Component).vm
    expect(vm.color).toEqual('white')
  })

  describe('Visuals', () => {
    test('defaults', () => {
      // Use mount and snapshot assertions
      const wrapper = mount(defaultStory(), {
        localVue: createLocalVue().localVue
      })

      expect(wrapper.isVueInstance()).toBe(true)
      expect(wrapper.element).toMatchSnapshot()
    })
    test('alternative colors', () => {
      // Use mount and snapshot assertions
      const wrapper = mount(alternativeColor(), {
        localVue: createLocalVue().localVue
      })

      expect(wrapper.isVueInstance()).toBe(true)
      expect(wrapper.element).toMatchSnapshot()
    })
  })
})
