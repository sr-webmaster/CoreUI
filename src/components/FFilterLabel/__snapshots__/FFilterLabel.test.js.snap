// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FFilterLabel Visuals alternative colors 1`] = `
<div
  class="black--text"
>
  <span
    class="no-wrap font-weight-bold"
  >
    Label
  </span>
</div>
`;

exports[`FFilterLabel Visuals defaults 1`] = `
<div
  class="white--text"
>
  <span
    class="no-wrap font-weight-bold"
  >
    Label
  </span>
</div>
`;
